lib/hls/hls_parser.js

```diff
@@ -614,9 +614,23 @@ shaka.hls.HlsParser = class {
       if (playlist.segments.length) {
         const parsedUri = new goog.Uri(playlist.segments[0].absoluteUri);
         const extension = parsedUri.getPath().split('.').pop();
-        const mimeType = HlsParser.RAW_FORMATS_TO_MIME_TYPES_[extension];
+        let mimeType = HlsParser.RAW_FORMATS_TO_MIME_TYPES_[extension];
         if (mimeType) {
           fullMimeType = mimeType;
+        } else if (extension === 'ts') {
+          // TODO: Fetch one segment a use the TsParser to analize if there is
```

Comment:
:sob:

Reply:
Do you prefer any other way?

Reply:
I don't love fetching segments for analysis.  We quit doing it for timestamps, and I don't want to go back.

This is probably not as bad, since it's a media playlist, so we will only ever fetch one segment.  I still wish we had another way.  But Apple didn't design HLS for MediaSource, so it's tough to make things like this work without probing into the segments.